pkg/plugins/runtime/k8s/controllers/gatewayapi/secret_convertion.go

```diff
@@ -64,21 +55,21 @@ func (r *GatewayReconciler) createSecretIfMissing(
 	return resKey, nil
 }
 
+// convertSecret returns the data to be stored as a Kuma secret or an error to
+// be displayed in a condition message.
 func convertSecret(secret *kube_core.Secret) ([]byte, error) {
 	if secret.Type != kube_core.SecretTypeTLS {
-		return nil, errors.Errorf("only secrets of type %q are supported", secret.Type)
+		return nil, errors.Errorf("only secrets of type %q are supported", kube_core.SecretTypeTLS)
 	}
 
-	privatePEM := pem.EncodeToMemory(&pem.Block{
```

Comment:
@jakubdyszkiewicz it looks like the assumption about `tls.key`/`tls.cert` for `kubernetes.io/tls` `Secrets` across k8s projects is that the data in already in PEM format. The Gateway API conformance tests were failing because of it. Do you think it makes sense to change this accordingly?

Reply:
I'm resolving this, there's no way around it :slightly_smiling_face: